python/ray/serve/api.py

```diff
@@ -453,6 +453,9 @@ def deployment(
             "_autoscaling_config is provided."
         )
 
+    if num_replicas == 0:
+        raise ValueError("num_replicas is expected to larger than 0")
```

Comment:
this conflict with the `NonNegativeInt` requirement you changed in DeploymentConfig
https://github.com/sihanwang41/ray/blob/219997e449/python/ray/serve/config.py#L111

which one is true? we shouldn't have two validation pass that disagree with each other

Reply:
This is because we are using the DeploymentConfig for autoscaling and non autoscaling. If we are okay to receive num_replicas == 0 for non autoscaling, I can remove it.

Reply:
That makes sense to me. In the future, we may want to introduce a new field or refactor the codebase, so `num_replicas` isn't so overloaded. Right now it represents both the user-configured `num_replicas` as well as the current number of replicas set by autoscaling.